openedx/core/djangoapps/appsembler/auth/models.py

```diff
@@ -0,0 +1,28 @@
+from django.db import models
+
+from oauth2_provider.models import Application
+
+
+class TrustedApplication(models.Model):
+    """Identify that an OAuth2 application is a trusted application.
+
+    This exists to replace behavior that existed in Hawthorn but apparently
+    removed in subsequent releases. The model class that existed in Hawthorn was
+    `edx_oauth2_provider.models.TrustedClient`. Since the actors in the
+    participating Python OAuth2 packages changed from 'Client' to 'Application',
+    we name this class `TrustedApplication` instead of `TrustedClient`.
+
+    This functionality _might_ exist in Juniper. However it is not clear, we
+    have our own customizations for auth, so adding this model let's us reuse
+    some of the same mechanisms to prevent trusted application.
+
+    We could extend Application to add a 'trusted bit' to it and this might be
+    a good idea in the future. For now we're implementing this as a distinct
+    model to make it easier to debug and troubleshoot
+    """
+    application = models.ForeignKey(Application, on_delete=models.CASCADE)
```

Comment:
Optional: I think it makes sense to make it OneToOne or just add `unique=True`.

```suggestion
    application = models.ForeignKey(Application, on_delete=models.CASCADE, unique=True)
```

Reply:
Yeah, that makes sense. Good catch. Best I do that now as this is the cheapest time ever we can makes this a 1:1